pkg/router/router_test.go

```diff
@@ -0,0 +1,249 @@
+
+	routev1 "github.com/openshift/api/route/v1"
+
+	projectfake "github.com/openshift/client-go/project/clientset/versioned/fake"
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
+	routefake "github.com/openshift/client-go/route/clientset/versioned/fake"
+	routelisters "github.com/openshift/client-go/route/listers/route/v1"
+
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	kubefake "k8s.io/client-go/kubernetes/fake"
+
+	"k8s.io/klog"
+
+	"github.com/openshift/router/haproxy"
+	routercmd "github.com/openshift/router/pkg/cmd/infra/router"
+	"github.com/openshift/router/pkg/router"
+	"github.com/openshift/router/pkg/router/controller"
+	templateplugin "github.com/openshift/router/pkg/router/template"
+	"github.com/openshift/router/pkg/router/writerlease"
+)
+
+type harness struct {
+	routeClient routeclient.Interface
+
+	namespace string
+	uidCount  int
+}
+
+func (h *harness) nextUID() types.UID {
+	h.uidCount++
+	return types.UID(fmt.Sprintf("%03d", h.uidCount))
+}
+
+var h *harness
+
+func TestMain(m *testing.M) {
+	logFlags := flag.FlagSet{}
+	klog.InitFlags(&logFlags)
+	if err := logFlags.Set("v", "6"); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// Client stuff
+	client := kubefake.NewSimpleClientset()
+	routeClient := routefake.NewSimpleClientset()
+	projectClient := projectfake.NewSimpleClientset()
+
+	namespace := "default"
+
+	h = &harness{
+		routeClient: routeClient,
+		namespace:   namespace,
+	}
+
+	// Other shared junk
+	routerSelection := &routercmd.RouterSelection{}
+	factory := routerSelection.NewFactory(routeClient, projectClient.ProjectV1().Projects(), client)
+	informer := factory.CreateRoutesSharedInformer()
+	routeLister := routelisters.NewRouteLister(informer.GetIndexer())
+	lease := writerlease.New(time.Minute, 3*time.Second)
+	go lease.Run(wait.NeverStop)
+	tracker := controller.NewSimpleContentionTracker(informer, namespace, 60*time.Second)
+	tracker.SetConflictMessage(fmt.Sprintf("The router detected another process is writing conflicting updates to route status with name %q. Please ensure that the configuration of all routers is consistent. Route status will not be updated as long as conflicts are detected.", namespace))
+	go tracker.Run(wait.NeverStop)
+
+	var plugin router.Plugin
+
+	workdir, err := ioutil.TempDir("", "router")
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	fmt.Printf("router working directory: %s\n", workdir)
+
+	// The template plugin which is wrapped
+	svcFetcher := templateplugin.NewListWatchServiceLookup(client.CoreV1(), 60*time.Second, namespace)
+	pluginCfg := templateplugin.TemplatePluginConfig{
+		WorkingDir:            workdir,
+		DefaultCertificateDir: workdir,
+		ReloadFn:              func(shutdown bool) error { return nil },
+		Template:              string(haproxy.MustAsset("images/router/haproxy/conf/haproxy-config.template")),
```

Comment:
@openshift/sig-network-edge if nobody wants to switch to using bindata for this generally (i.e. eliminate the template path option entirely), I can revert the bindata stuff and rely on env or something (e.g. default relative to pwd) to configure the existing path variable. I doubt the bindata stuff is worth maintaining just for this test

Reply:
I went ahead and rolled it back